clinvar_ingest/api/model/requests.py

```diff
@@ -1,48 +1,120 @@
 from datetime import date, datetime
+from pathlib import PurePath
+from typing import Any, Callable, Union
 
-from pydantic import BaseModel, ConfigDict, Field
+from pydantic import (
+    AnyUrl,
+    BaseModel,
+    ConfigDict,
+    Field,
+    RootModel,
+    field_serializer,
+    validator,
+)
 
 
 def to_title_case(string: str) -> str:
     return " ".join(word.capitalize() for word in string.split("_"))
 
 
+def walk_and_replace(d: dict, dump_fn: Callable[[Any], Any]):
+    if isinstance(d, dict):
+        for k, v in d.items():
+            d[k] = walk_and_replace(v, dump_fn)
+    elif isinstance(d, list):
+        for i in range(len(d)):
+            d[i] = walk_and_replace(d[i], dump_fn)
+    else:
+        d = dump_fn(d)
+    return d
+
+
+def _dump_fn(val):
+    if isinstance(val, PurePath):
+        return str(val)
+    if isinstance(val, BaseModel):
+        return val.model_dump()
+    return val
+
+
 class ClinvarFTPWatcherRequest(BaseModel):
     model_config = ConfigDict(
         alias_generator=to_title_case,
         populate_by_name=True,
         json_schema_extra={
             "examples": [
                 {
+                    "Host": "https://ftp.ncbi.nlm.nih.gov",
+                    "Directory": "/pub/clinvar/xml/clinvar_variation/weekly_release",
                     "Name": "ClinVarVariationRelease_2023-1104.xml.gz",
                     "Size": 3160398711,
                     "Released": "2023-11-05 15:47:16",
                     "Last Modified": "2023-11-05 15:47:16",
-                    "Directory": "/pub/clinvar/xml/clinvar_variation/weekly_release",
                     "Release Date": "2023-11-04",
                 }
             ]
         },
     )
 
+    host: AnyUrl
+    directory: str
     name: str
     size: int
     released: datetime
     last_modified: datetime
-    directory: str
     release_date: date
 
 
-class ParseRequest(BaseModel):
+class CopyResponse(BaseModel):
     ftp_path: str
     gcs_path: str
+
+
+class ParseRequest(BaseModel):
+    input_path: str
+    output_path: str
     no_disassemble: bool = Field(default=True)
     no_jsonify_content: bool = Field(default=True)
 
 
-class CopyResponse(BaseModel):
-    ftp_path: str
-    gcs_path: str
+class GCSBlobPath(RootModel):
+    """
+    A GCS blob path, such as gs://my-bucket/my-file.txt
+    Validates path structure, does not check if the file exists.
+    """
+
+    root: str
+
+    @validator("root")
+    def _validate(cls, v):  # pylint: disable=E0213
+        if not v.startswith("gs://"):
+            raise ValueError(f"Must be a gs:// URL: {v}")
+        AnyUrl(v)
+        return v
+
+
+class PurePathModel(RootModel):
```

Comment:
Does [filepath](https://docs.pydantic.dev/latest/api/types/#pydantic.types.FilePath) work for this?

Reply:
I was using FilePath originally but was having problems with serializing with `.model_dump()`, but now I'm having trouble replicating that issue. So I'll try to switch it back

Reply:
If I change the `parsed_files` Union to use `FilePath` instead of `PurePath` or `PurePathModel` and remove the custom field serializer, the output is just this, with all the paths being empty strings:
`{"parsed_files":{"gene":"","gene_association":"","variation":"","trait":"","trait_set":"","clinical_assertion":"","variation_archive":""}}`

https://github.com/clingen-data-model/clinvar-ingest/compare/325b9ae4d9d96eb45a17c97e4ca9c2de1cc8e414..29dd5fe7457a79dfa257036130cba394fa67973b